src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -297,6 +307,43 @@ mod test {
         assert_eq!(got, exp, "bit_length(NULL)");
     }
 
+    #[test]
+    fn test_oct_int() {
+        let cases = vec![
+            (Datum::I64(12), Datum::Bytes(b"14".to_vec())),
+            (Datum::I64(8), Datum::Bytes(b"10".to_vec())),
+            (Datum::I64(42), Datum::Bytes(b"52".to_vec())),
+            (Datum::I64(365), Datum::Bytes(b"555".to_vec())),
+            (Datum::I64(1024), Datum::Bytes(b"2000".to_vec())),
+            (Datum::Null, Datum::Null),
+            (
+                Datum::I64(i64::max_value()),
+                Datum::Bytes(b"777777777777777777777".to_vec()),
+            ),
+            (
+                Datum::I64(i64::min_value()),
+                Datum::Bytes(b"1000000000000000000000".to_vec()),
+            ),
+            (
+                Datum::I64(-1),
+                Datum::Bytes(b"1777777777777777777777".to_vec()),
+            ),
```

Comment:
What about add some corner cases?  For example:
```sql
mysql> select oct(18446744073709551616);
+---------------------------+
| oct(18446744073709551616) |
+---------------------------+
| 1777777777777777777777    |
+---------------------------+
1 row in set (0.00 sec)

mysql> select oct(-9223372036854775809);
+---------------------------+
| oct(-9223372036854775809) |
+---------------------------+
| 777777777777777777777     |
+---------------------------+
1 row in set (0.00 sec)

mysql> select oct(111111111111111111111111111111111111111111111111111111111111111);
+----------------------------------------------------------------------+
| oct(111111111111111111111111111111111111111111111111111111111111111) |
+----------------------------------------------------------------------+
| 1777777777777777777777                                               |
+----------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select oct(18446744073709551614);
+---------------------------+
| oct(18446744073709551614) |
+---------------------------+
| 1777777777777777777776    |
+---------------------------+
1 row in set (0.00 sec)

mysql> select oct(18446744073709551615);
+---------------------------+
| oct(18446744073709551615) |
+---------------------------+
| 1777777777777777777777    |
+---------------------------+
1 row in set (0.00 sec)

```

Reply:
I feel strange about  `111111111111111111111111111111111111111111111111111111111111111`, 
`#[allow(overflowing_literals)]` also cannot make it compile.
And `18446744073709551615` is Contradicted with one of my test data.

Reply:
```sql
+---------------------------+
| oct(18446744073709551616) |
+---------------------------+
| 1777777777777777777777    |
+---------------------------+
```
```sql
+---------------------------+
| oct(18446744073709551615) |
+---------------------------+
| 1777777777777777777777    |
+---------------------------+
```
Are they not contradictory?

Reply:
@yjhmelody  yep, that's why we should test the corner cases, the behavior of ` format! ` is not always as we expected..

Reply:
I means why oct(18446744073709551616) == oct(18446744073709551615)
And ...
How it can do that

OCT(N)
- Returns a string representation of the octal value of N, where N is a longlong (BIGINT) number. This is equivalent to CONV(N,10,8). Returns NULL if N is NULL.

Reply:
@yjhmelody Let's take a look into MySQL source code.  `oct` is implemented in [`sql/item_create.cc`](https://github.com/mysql/mysql-server/blob/913071c0b16cc03e703308250d795bc381627e37/sql/item_create.cc#L6784), which direct us to [`sql/item_strfunc.cc`](https://github.com/mysql/mysql-server/blob/913071c0b16cc03e703308250d795bc381627e37/sql/item_strfunc.cc#L3929 ).
Line 3966 is where the input string is converted to int:
```c++
      dec= (longlong) my_strntoull(res->charset(), res->ptr(), res->length(),
                                   from_base, &endptr, &err);
```
Following the function we can trace down to [`strings/ctype-simple.c`](https://github.com/mysql/mysql-server/blob/b93c1661d689c8b7decc7563ba15f6ed140a4eb6/strings/ctype-simple.cc#L1221):
```c++
  for (; shift > 0; shift--, ull *= 10) /* Left shift */
  {
    if (ull > CUTOFF) goto ret_too_big; /* Overflow, number too big */
  }
```

```c++
ret_too_big:
  *endptr = (char *)str;
  *error = MY_ERRNO_ERANGE;
  return unsigned_flag ? ULLONG_MAX
                       : negative ? (ulonglong)LLONG_MIN : (ulonglong)LLONG_MAX;
}
```
These parts indicate that if this function return INT_MAX/INT_MIN if the input is overflow. 
so oct(18446744073709551616) == oct(18446744073709551615) = oct(1111111111111111111111111111111111111111111111).